core/templates/dev/head/galleries/Gallery.js

```diff
@@ -99,8 +99,10 @@ oppia.factory('searchService', [
         _lastSelectedCategories = angular.copy(selectedCategories);
         _lastSelectedLanguageCodes = angular.copy(selectedLanguageCodes);
         _searchCursor = data.search_cursor;
+
         $rootScope.$broadcast(
           'refreshGalleryData', data, hasPageFinishedLoading());
+        $translate.refresh();
```

Comment:
Yes, this is the only place. The call to refresh() is necessary to translate the new explorations. I added the comment.